components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/event/OAuthEventInterceptor.java

```diff
@@ -226,4 +228,32 @@ default void onPostTokenRevocationBySystem(AccessTokenDO accessTokenDO, Map<Stri
             throws IdentityOAuth2Exception {
 
     }
+
+    /**
+     * This will be called before when Tokens Revoked by application.
+     *
+     * @param revokeRequestDTO {@link OAuthAppRevocationRequestDTO}
+     * @param params           Additional parameters
+     * @throws IdentityOAuth2Exception
+     */
+    default void onPreTokenRevocationByApplication(OAuthAppRevocationRequestDTO revokeRequestDTO,
+                                                   Map<String, Object> params) throws IdentityOAuth2Exception {
+
+    }
+
+    /**
+     * This will be called after when Tokens Revoked by application.
+     *
+     * @param revokeRequestDTO  {@link OAuthAppRevocationRequestDTO}
+     * @param revokeResponseDTO
+     * @param accessTokenDOs    {@link AccessTokenDO}
+     * @param params            Additional parameters
+     * @throws IdentityOAuth2Exception
+     */
+    default void onPostTokenRevocationByApplication(
+            OAuthAppRevocationRequestDTO revokeRequestDTO,
+            org.wso2.carbon.identity.oauth.dto.OAuthRevocationResponseDTO revokeResponseDTO,
```

Comment:
any reason behind using this class instead of https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dto/OAuthRevocationResponseDTO.java?

Reply:
oauth.dto.OAuthRevocationResponseDTO is used in order to be consistent with the existing authorized-apps API implantation.